setup.py

```diff
@@ -74,7 +74,7 @@
         'dev': [
             "cram >=0.7",
             "deepdiff >=4.3.2, <8.0.0",
-            "flake8",
+            "flake8 >=7.0.0, <8",
```

Comment:
The error which prompted this PR was due to 7.1.2 to 7.2.0 bump, so wouldn't this pin not have prevented the CI failure and thus keep the door open to future similar issues?

Reply:
Yes, but I think it's still useful to have new flake8 versions. Note that Nextstrain CLI has [no upper pin on most dev deps](https://github.com/nextstrain/cli/blob/f4ee3ceb525b2b4ea2d7a1fcaa7bd06a1582d034/setup.py#L138-L156), including flake8. But maybe, related to #1557, we should pin exact and update ~regularly?

Reply:
I agree with the desire to keep current for deps such as these. But having CI fail like it did in https://github.com/nextstrain/augur/pull/1756#issuecomment-2774401930 isn't good for development and should be avoided. A better approach would be a dependebot-like PR for flake8 upgrades.

Reply:
This is well-suited for #1538. I'll try to revisit that PR sometime.